src/Numerics/Mesh/Filters.jl

```diff
@@ -141,27 +141,62 @@ struct ExponentialFilter <: AbstractSpectralFilter
     "filter matrix"
     filter::Any
 
-    function ExponentialFilter(
-        grid,
-        Nc = 0,
-        s = 32,
-        α = -log(eps(eltype(grid))),
-    )
-        AT = arraytype(grid)
-        # XXX: Needs updating for multiple polynomial orders
-        N = polynomialorders(grid)
-        # Currently only support single polynomial order
-        @assert all(N[1] .== N)
-        N = N[1]
-        ξ = referencepoints(grid)[1]
+    if dg.diffusion_direction isa EveryDirection ||
+        dg.diffusion_direction isa HorizontalDirection
+
+        function ExponentialFilter(
+            grid,
+            Nc = 0,
+            s = 32,
+            α = -log(eps(eltype(grid))),
+            direction = HorizontalDirection(),
+        )
+            AT = arraytype(grid)
+            # Tuple of polynomial degrees (N₁, N₂, N₃)
+            N = polynomialorders(grid)
+            # In 2D, we assume same polynomial order in the horizontal
+            dim = dimensionality(grid)
+            @assert dim == 2 || N[1] == N[2]
+            horiz_polyorder = N[1]
+            horiz_ξ = referencepoints(grid)[1]
 
-        @assert iseven(s)
-        @assert 0 <= Nc <= N
+            @assert iseven(s)
+            @assert 0 <= Nc <= horiz_polyorder
 
-        σ(η) = exp(-α * η^s)
-        filter = spectral_filter_matrix(ξ, Nc, σ)
+            σ(η) = exp(-α * η^s)
 
-        new(AT(filter))
+            #TODO: think about how this matrix changes in different directions
+            filter = spectral_filter_matrix(horiz_ξ, Nc, σ)
+
+            new(AT(filter))
+        end
+    end
+
+    if dg.diffusion_direction isa EveryDirection ||
+        dg.diffusion_direction isa VerticalDirection
+
+        function ExponentialFilter(
+            grid,
+            Nc = 0,
+            s = 32,
+            α = -log(eps(eltype(grid))),
+            direction = VerticalDirection(),
+        )
+            AT = arraytype(grid)
+            # Tuple of polynomial degrees (N₁, N₂, N₃)
+            vert_polyorder = N[end]
+            vert_ξ = referencepoints(grid)[end] 
+            
+            @assert iseven(s)
+            @assert 0 <= Nc <= vert_polyorder
+
+            σ(η) = exp(-α * η^s)
+
+            #TODO: think about how this matrix changes in different directions
+            filter = spectral_filter_matrix(vert_ξ, Nc, σ)
+
+            new(AT(filter))
+        end
```

Comment:
@mwarusz if I understoon your suggestion here, was to base this similar to what was done in #1656 . In that PR, the direction was queried from the diffusion direction. But I am afraid this will be wrong here.